pbf2json.go

```diff
@@ -574,58 +573,47 @@ func bytesToLatLon(data []byte) map[string]string {
 
 // encode a node as bytes (between 12 & 13 bytes used)
 func nodeToBytes(node *osmpbf.Node) (string, []byte) {
+	stringid := strconv.FormatInt(node.ID, 10)
 
-	var bufval bytes.Buffer
-
+	buf := make([]byte, 14)
 	// encode lat/lon as 64 bit floats packed in to 8 bytes,
 	// each float is then truncated to 6 bytes because we don't
 	// need the additional precision (> 8 decimal places)
 
-	var latBytes = make([]byte, 8)
-	binary.BigEndian.PutUint64(latBytes, math.Float64bits(node.Lat))
-	bufval.Write(latBytes[0:6])
-
-	var lonBytes = make([]byte, 8)
-	binary.BigEndian.PutUint64(lonBytes, math.Float64bits(node.Lon))
-	bufval.Write(lonBytes[0:6])
+	binary.BigEndian.PutUint64(buf, math.Float64bits(node.Lat))
+	binary.BigEndian.PutUint64(buf[6:], math.Float64bits(node.Lon))
 
 	// generate a bitmask for relevant tag features
-	var isEntrance = isEntranceNode(node)
-	if isEntrance > 0 {
-		// leftmost two bits are for the entrance, next two bits are accessibility
-		// remaining 4 rightmost bits are reserved for future use.
-		var bitmask = isEntrance << 6
-		var isWheelchairAccessible = isWheelchairAccessibleNode(node)
-		if isWheelchairAccessible > 0 {
-			bitmask |= isWheelchairAccessible << 4
-		}
-		bufval.WriteByte(bitmask)
+	isEntrance := isEntranceNode(node)
+	if isEntrance == 0 {
+		return stringid, buf[:12]
 	}
 
-	stringid := strconv.FormatInt(node.ID, 10)
-	byteval := bufval.Bytes()
+	// leftmost two bits are for the entrance, next two bits are accessibility
+	// remaining 4 rightmost bits are reserved for future use.
+	bitmask := isEntrance << 6
+	bitmask |= isWheelchairAccessibleNode(node) << 4
+	buf[12] = bitmask
 
-	return stringid, byteval
+	return stringid, buf[:13]
 }
 
 func idSliceToBytes(ids []int64) []byte {
-	var bufval bytes.Buffer
-	for _, id := range ids {
-		var idBytes = make([]byte, 8)
-		binary.BigEndian.PutUint64(idBytes, uint64(id))
-		bufval.Write(idBytes)
+	buf := make([]byte, 8*len(ids))
+	for i, id := range ids {
+		binary.BigEndian.PutUint64(buf[8*i:], uint64(id))
 	}
-	return bufval.Bytes()
+	return buf
 }
 
 func bytesToIDSlice(bytes []byte) []int64 {
-	var ids []int64
 	if len(bytes)%8 != 0 {
 		log.Fatal("invalid byte slice length: not divisible by 8")
-		return make([]int64, 0)
 	}
-	for i := 0; i < len(bytes)/8; i++ {
-		ids = append(ids, int64(binary.BigEndian.Uint64(bytes[i*8:(i*8)+8])))
+
+	ids := make([]int64, 0, len(bytes)/8)
```

Comment:
should this be `make([]int64, len(bytes)/8)`, ie. setting the `len` instead of the `cap`?
again yak shaving but that will avoid use of `append` since we can use `ids[i]` instead.

Reply:
I made both changes. This made a difference 8% not sure exactly way. The above did not.